stl/inc/numbers

```diff
@@ -0,0 +1,188 @@
+    template <floating_point _Floating>
+    inline constexpr _Floating phi_v<_Floating> = static_cast<_Floating>(1.618033988749895);
+#else // ^^^ defined(__cpp_lib_concepts) / !defined(__cpp_lib_concepts) vvv
+    template <>
+    inline constexpr double e_v<double> = static_cast<double>(2.718281828459045);
+    template <>
+    inline constexpr double log2e_v<double> = static_cast<double>(1.4426950408889634);
+    template <>
+    inline constexpr double log10e_v<double> = static_cast<double>(0.4342944819032518);
+    template <>
+    inline constexpr double pi_v<double> = static_cast<double>(3.141592653589793);
+    template <>
+    inline constexpr double inv_pi_v<double> = static_cast<double>(0.3183098861837907);
+    template <>
+    inline constexpr double inv_sqrtpi_v<double> = static_cast<double>(0.5641895835477563);
+    template <>
+    inline constexpr double ln2_v<double> = static_cast<double>(0.6931471805599453);
+    template <>
+    inline constexpr double ln10_v<double> = static_cast<double>(2.302585092994046);
+    template <>
+    inline constexpr double sqrt2_v<double> = static_cast<double>(1.4142135623730951);
+    template <>
+    inline constexpr double sqrt3_v<double> = static_cast<double>(1.7320508075688772);
+    template <>
+    inline constexpr double inv_sqrt3_v<double> = static_cast<double>(0.5773502691896257);
+    template <>
+    inline constexpr double egamma_v<double> = static_cast<double>(0.5772156649015329);
+    template <>
+    inline constexpr double phi_v<double> = static_cast<double>(1.618033988749895);
+
+    template <>
+    inline constexpr float e_v<float> = static_cast<float>(e_v<double>);
+    template <>
+    inline constexpr float log2e_v<float> = static_cast<float>(log2e_v<double>);
+    template <>
+    inline constexpr float log10e_v<float> = static_cast<float>(log10e_v<double>);
+    template <>
+    inline constexpr float pi_v<float> = static_cast<float>(pi_v<double>);
+    template <>
+    inline constexpr float inv_pi_v<float> = static_cast<float>(inv_pi_v<double>);
+    template <>
+    inline constexpr float inv_sqrtpi_v<float> = static_cast<float>(inv_sqrtpi_v<double>);
+    template <>
+    inline constexpr float ln2_v<float> = static_cast<float>(ln2_v<double>);
+    template <>
+    inline constexpr float ln10_v<float> = static_cast<float>(ln10_v<double>);
+    template <>
+    inline constexpr float sqrt2_v<float> = static_cast<float>(sqrt2_v<double>);
+    template <>
+    inline constexpr float sqrt3_v<float> = static_cast<float>(sqrt3_v<double>);
+    template <>
+    inline constexpr float inv_sqrt3_v<float> = static_cast<float>(inv_sqrt3_v<double>);
+    template <>
+    inline constexpr float egamma_v<float> = static_cast<float>(egamma_v<double>);
+    template <>
+    inline constexpr float phi_v<float> = static_cast<float>(phi_v<double>);
+
+    template <>
+    inline constexpr long double e_v<long double> = static_cast<long double>(e_v<double>);

```

Comment:
`static_cast<long double>` is unnecessary; it is always safe to widen from `double` to `long double` implicitly (especially on MSVC's ABI, where they have the exact same representation).